src/cli/mod.rs

```diff
@@ -518,6 +529,42 @@ fn installed_after_release(next: &str) -> bool {
     false
 }
 
+fn notification_message(current: &str, next: &str) -> (String, String) {
+    match EolDate::fetch() {
+        Ok(eol) => {
+            let (y, m, d) = eol.ymd;
+            match eol.status() {
+                EolStatus::Exceeded => {
+                    return (
+                        fomat!("Support for Pop!_OS " (current) " has ended"),
+                        fomat!(
+                            "Security and application updates are no longer provided "
+                            "for Pop!_OS " (current) ". Upgrade to Pop!_OS " (next)
+                            " to keep your computer secure."
```

Comment:
I am not seeing this code path executed when the date is beyond that of the EOL date.

Reply:
This is perplexing because the code that returns this is:

```rust
let (year, month, day) = self.ymd;
let eol = ymd_to_utc(year as i32, month, day);
let current = Utc::now().date();

if current >= eol {
    EolStatus::Exceeded
} else if imminent(current, eol) {
    EolStatus::Imminent
} else {
    EolStatus::Ok
}
```

If the current date is greater than, or equal to, the eol date; it should be returning `Exceeded`.

Reply:
Would it be feasible to add tests for that function?

Reply:
Tests can be added to the eol module to test the fetch method, yes.